sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -83,7 +83,8 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
   }
 
   {
-    std::shared_lock<std::shared_timed_mutex> Lock(MGraphLock);
+    std::unique_lock<std::shared_timed_mutex> Lock(MGraphLock, std::defer_lock);
+    lockSharedTimedMutex(Lock);
```

Comment:
Why?
We only need a read-lock to enqueue smth as it's non-modifying operation.

Reply:
We have already a test for the test case, which sporadically fails without that on Windows and Linux.
About a couple of weeks ago, we had a single lock guard in the function for NewCmd and enqueue, but now, test fails when you try to submit to a single queue by multi-threads. 
This patch fixes the issue.

Reply:
OK. You're going to serialize enqueue operations, aren't you? Then, I believe there should be a distinct lock for it. Justification: there are other places where we acquire read-lock and enqueue smth or do smth.

As far as I understand, in general there are two operation for graph: put command to graph and enqueue. We can't put multiple commands to graph simultaneously as it's write operation. Though we should be able to enqueue multiple commands for execution, isn't it?

Reply:
Yes, I am going to serialize enqueue operations, because without it we can face with fails when try to submit to a single queue by multiple threads. Thanks, I will add a separate lock for that.

Reply:
It was decided to use a single lock. see conversation below.